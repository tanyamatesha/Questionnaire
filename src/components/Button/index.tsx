import React from "react"
import styled, { css } from "styled-components"

type ButtonSizing = "md" | "lg" | "stretch"
type ButtonProps = Omit<React.HTMLProps<HTMLButtonElement>, "width"> & {
  sizing?: ButtonSizing
}

const SIZING_CSS = css<{ sizing?: ButtonSizing }>`
  ${props => {
    switch (props.sizing) {
      case "lg":
        return `padding: 0.75rem 1.5rem;`
      case "stretch":
        return `
              width: 100%;
              padding: 0.75rem 0;
              `
      case "md":
      default:
        return `padding: 0.625rem 1.5rem;`
    }
  }}
`

const Button = styled.button<ButtonProps>`
  display: inline-block;
  width: fit-content;
  white-space: nowrap;

  ${SIZING_CSS}

  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.4375rem;

  border-radius: 0.25rem;
  border-width: 0;

  text-align: center;
  text-decoration: none;

  margin-top: 1.5rem;
  margin-bottom: 1rem;

  background: #CDC047;
  color: #1B1B1B;

  &:hover, &:focus {
    cursor: pointer;
    background: #BAA90E;
    outline: none;
  }

  &:active {  
    background: #BAA90E;
    box-shadow: inset 0.25rem 0.25rem 0.75rem #8B7E08;
  }
`

Button.defaultProps = {
  type: "button",
  sizing: "md"
}

export default Button
