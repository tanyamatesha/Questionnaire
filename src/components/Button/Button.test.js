
import * as React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'

import Button from 'components/Button/index.tsx'

describe('Button', () => {
  describe('type prop', () => {
    const typeOptions = ["submit", "reset", "button"]

    typeOptions.forEach(typeOption =>
      it(`should render element with ${typeOption} type`, () => {
        render(<Button type={typeOption} />)
        const button = screen.getByRole('button')

        expect(button).toHaveAttribute('type', typeOption)
      })
    )
  })

  describe('size prop', () => {
    const sizeOptions = ["md", "lg", "stretch"]

    sizeOptions.forEach(sizeOption =>
      it(`should render element with ${sizeOption} size className`, () => {
        render(<Button sizing={sizeOption} />)
        const button = screen.getByRole('button')

        expect(button).toMatchSnapshot()
      })
    )
  })
})
