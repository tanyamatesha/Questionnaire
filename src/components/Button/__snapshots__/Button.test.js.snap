// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button size prop should render element with lg size 1`] = `
<button
  class="sc-bdfBwQ eDWGdu"
  type="button"
/>
`;

exports[`Button size prop should render element with lg size className 1`] = `
<button
  class="sc-bdfBwQ eDWGdu"
  type="button"
/>
`;

exports[`Button size prop should render element with md size 1`] = `
<button
  class="sc-bdfBwQ iSFsKt"
  type="button"
/>
`;

exports[`Button size prop should render element with md size className 1`] = `
<button
  class="sc-bdfBwQ iSFsKt"
  type="button"
/>
`;

exports[`Button size prop should render element with stretch size 1`] = `
<button
  class="sc-bdfBwQ iSWvtD"
  type="button"
/>
`;

exports[`Button size prop should render element with stretch size className 1`] = `
<button
  class="sc-bdfBwQ iSWvtD"
  type="button"
/>
`;
